test/api/controller/base.test.js

```diff
@@ -633,9 +635,58 @@ describe('#base/native controller', () => {
       should(searchBody).be.eql({ query: { foo: 'bar' } });
     });
 
+    it('should extract search params when invoking the route with GET', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'GET' } } };
+      request.input.body = null;
+      const
+        { from, query, scrollTTL, searchBody, size } = nativeController.getSearchParams(request);
+
+      should(from).be.eql(1);
+      should(size).be.eql(11);
+      should(scrollTTL).be.eql('10m');
+      should(query).be.eql({});
+      should(searchBody).be.eql({ query: { foo: 'bar' } });
+    });
+
+    it('should extract searchBody param when the route is invoked using GET', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'GET' } } };
+      request.input.body = null;
+      const searchBody = nativeController.getSearchBody(request);
+
+      should(searchBody).be.eql({ query: { foo: 'bar' } });
+    });
+
+    it('should extract searchBody param', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'POST' } } };
+      const searchBody = nativeController.getSearchBody(request);
+
+      should(searchBody).be.eql({ query: { foo: 'bar' } });
+    });
+
+    it('should throw when the route is invoked with GET and invalid search body is provided', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'GET' } } };
+      request.input.body = null;
+      request.input.args.searchBody = {};
+
+      should(() => {
+        nativeController.getSearchBody(request);
+      }).throw({ id: 'api.assert.invalid_type', message: 'Wrong type for argument "searchBody" (expected: string)' });
+    });
+
+    it('should throw when the route is invoked with GET and no search body is provided', () => {
```

Comment:
Should it? When I send a `document:search` without a null body with the current API, the search matches all documents, same as with an empty body. We should keep the same behavior with the `GET` verb.